types/group.js

```diff
@@ -0,0 +1,62 @@
+const GroupKeys = require('./groupKeys')
+const Serializable = require('./serializable')
+
+class Group extends Serializable {
+  constructor(data, groupKeys) {
```

Comment:
I'm not quite in line with the types we have defined here. I think we're going to run into some problems going forward and will be forced to refactor later.

Let's try to split this up a bit and think about what concept each type maps to (not saying this is definitely the way to go, just what is making sense in my mind as I think this through).

- **Group**: When I think of a `Group` object, I think of the data returned from a `GroupInfo()` request. Name, public key, group ID, possibly some other metadata. This will be generic no matter which group member is fetching it.
- **Membership(Meta?)**: This could be a user's connection to a group -- containing their group capabilities, their membership key definition, other group-client metadata.
- **MembershipKey**: (Basically what you have for GroupKeys?). This could be a handy object, though I suggest we call it `MembershipKey since that we what we've called the encrypted group private key?

This is not much different the the types you have, but conceptually I'm not sure a Group should take a Membership as part of it's constructor -- Perhaps it is the Membership type which contains a Group instead, to make easy access to Group info from a client's Membership to that group?

Plenty open to other thought's opinions here.